llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -148,6 +148,20 @@ static ShaderStage sourceLangToShaderStage(SpvGenStage sourceLang) {
   }
 }
 
+// =====================================================================================================================
+// Disassemble SPIR-V binary file.
+//
+// @param binSize: Binary size.
+// @param code: Pointer to binary.
+// @param header: Extra info.
+static void disassembleSpirv(unsigned binSize, const void *code, const llvm::Twine &header) {
+  unsigned textSize = binSize * 10 + 1024;
+  std::vector<char> spvText(textSize);
+  LLPC_OUTS("\nSPIR-V disassembly for " << header << ":\n");
+  spvDisassembleSpirv(binSize, code, textSize, spvText.data());
+  LLPC_OUTS(spvText.data() << "\n");
+}
```

Comment:
We unconditionally perform work (generate disassembly), but then conditionally use it (output if `EnableOuts` is active). Calling this function otherwise would silently create useless extra work.

So we should either exit early if `EnableOuts()` is not set, or add an assertion that `EnableOuts()` is set. I prefer the latter.

Reply:
Assertion added.